coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -58,14 +58,25 @@ class ShardKeyRegexPlanner(val dataset: Dataset,
     val nonMetricShardKeyFilters =
       LogicalPlan.getNonMetricShardKeyFilters(logicalPlan, dataset.options.nonMetricShardColumns)
     if (isMetadataQuery(logicalPlan)
-      || LogicalPlan.hasShardKeyEqualsOnly(logicalPlan, dataset.options.nonMetricShardColumns)) {
+      || (hasRequiredShardKeysPresent(nonMetricShardKeyFilters, dataset.options.nonMetricShardColumns) &&
```

Comment:
difficult to follow the logic
if (A || B && C)
 
 (a || B ) && C?
or 
A || (B && C) ?

Reply:
>A || (B && C) ?
if its metadata query OR (has all the required nonMetric ShardKeys (`_ws_`, `_ns_`) present with Equals filter )

Reply:
got you, got you!